tsa-core/src/main/kotlin/org/usvm/machine/interpreter/TvmDictOperationInterpreter.kt

```diff
@@ -1806,54 +2325,67 @@ class TvmDictOperationInterpreter(
 
         assertDictKeyLength(scope, dictCellRef, keyLength)
             ?: return
-        assertDictIsNotEmpty(scope, dictCellRef, dictId)
-            ?: return
 
         val keySort = ctx.mkBvSort(keyLength.toUInt())
-        val resultElement = scope.calcOnStateCtx { makeSymbolicPrimitive(keySort) }
-        val resultElementExtended = extendDictKey(resultElement, keyType)
 
         // since these entries were stored during execution, value overflow constraints have already been asserted
         val allSetEntries =
             scope.calcOnStateCtx {
                 memory.setEntries(dictCellRef, dictId, keySort, DictKeyInfo)
             }
+        val isInput = checkDictIsInput(scope, dictCellRef, allSetEntries)
+        if (isInput) {
+            doNextPrevInstOnInputDict(
+                dictCellRef = dictCellRef,
+                scope = scope,
+                keySort = keySort,
+                keyKind = keyKind,
+                key = key,
+                mode = mode,
+                allowEq = allowEq,
+                dictId = dictId,
+                valueType = valueType,
+                inst = inst,
+            )
+            return
+        }
 
-        // TODO input keys are not considered
         val storedKeys =
```

Comment:
I think that processing of allocated dict should also be moved to a separate function (that applies to all dict operations).